### PR TITLE
Provide a swipe navigation distance setting for users to config along with swipe indicator

### LINKED ISSUE #9615: Swipe Navigation Sensitivity Setting
## Test plan

https://github.com/brave/browser-laptop/pull/9617#issue-237356717

----

screen shot 2017-06-20 at 10 42 46 am
similar to system setting

### PR DESCRIPTION
**Pull request by darkdh**
fixes #9615 

Auditors: @cezaraugusto, @bbondy, @bradleyrichter

Test Plan:
1. Adjust swipe navigation sesitivity setting in
about:preferences#advanced
2. And swipe to navigate by different settings

It shouldn't show up on linux and window

Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

---
screen shot 2017-06-29 at 11 05 22 am

Also add swipe indicator

kapture 2017-06-29 at 11 06 41

### CODE REVIEW

**Comment by bradleyrichter** [comment]
@darkdh I think the slowest setting greatly improves on the accidental swipe issue we are having now. It felt natural as well. So I don't think the slider is necessary right now but keep it for other future needs?

**Comment by darkdh** [comment]
@bradleyrichter, we can make the slowest setting default but some users like swiping faster or even slower. We can't satisfy all users with one setting so we allow them to customize.
I'm a fast swiper and I believe @diracdeltas is even faster.

**Comment by darkdh** [comment]
also fix 2477

**Comment by cezaraugusto** [comment]
feature works but sometimes instead of swiping to the next/previous page it opens the context menu. I don't have proper str

**Comment by cezaraugusto** [comment]
which is tricky given it also happens with Chrome

**Comment by cezaraugusto** [comment]
btw what's the take on https://github.com/brave/browser-laptop/pull/9617#issuecomment-309923989 @bradleyrichter @darkdh ?

**Comment by darkdh** [comment]
@cezaraugusto it is addressed per discussion I had with @bradleyrichter yesterday
This PR is ready for review again.

File: app/renderer/components/navigation/navigator.js [hunk 1]
```diff
@@ -190,6 +192,18 @@ class Navigator extends React.Component {
     props.isWideURLbarEnabled = getSetting(settings.WIDE_URL_BAR)
     props.showNavigationBar = activeFrameKey !== undefined &&
       state.get('siteSettings') !== undefined
+    props.swipeLeftPercent = swipeLeftPercent ? (swipeLeftPercent + 1) * 1.2 : 1
+    props.swipeRightPercent = swipeRightPercent ? (swipeRightPercent + 1) * 1.2 : 1
+    // 0.85 is the default button opacity in less/navigationBar.less
+    // Remove this magic number once we migrate to Aphrodite
```
Comment (cezaraugusto) [inline app/renderer/components/navigation/navigator.js]: thanks for the comment

File: app/renderer/components/preferences/advancedTab.js [hunk 2]
```diff
@@ -35,6 +35,30 @@ class AdvancedTab extends ImmutableComponent {
...
+  get swipeNavigationDistanceSetting () {
+    if (platformUtil.isDarwin()) {
+      return <div>
+        <DefaultSectionTitle data-l10n-id='swipeNavigationDistance' />
+        <SettingsList listClassName={css(styles.swipeNavigation)}>
+          <span data-l10n-id='short' className={css(styles.swipeNavigation__shortLabel)} />
+          <input type='range' min='1' max='201' step='50' list='swipeDistance'
+            value={getSetting(settings.SWIPE_NAV_DISTANCE, this.props.settings)}
+            onChange={changeSetting.bind(null, this.props.onChangeSetting, settings.SWIPE_NAV_DISTANCE)} />
+          <datalist id='swipeDistance'>
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/advancedTab.js]: the front-end gods salute you for this great element ++ 😜

File: app/renderer/components/preferences/advancedTab.js [hunk 3]
```diff
@@ -80,6 +106,19 @@ const styles = StyleSheet.create({
...
+    alignItems: 'center'
+  },
+
+  swipeNavigation__shortLabel: {
+    marginRight: '5px'
+  },
+
+  swipeNavigation__longLabel: {
+    marginLeft: '5px'
+  },
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/advancedTab.js]: super cool

File: app/renderer/components/navigation/navigator.js (hunk 1 above)
Comment (bsclifton) [inline app/renderer/components/navigation/navigator.js]: ++

**Pull request merged**